### PR TITLE
forge(fmt): docs & contribution guide

### PR DESCRIPTION
**Pull request by rkrasiuk**
Expand the existing `README.md` covering the `fmt` development process and contribution guidelines

### CODE REVIEW

File: fmt/README.md [hunk 1]
```diff
@@ -215,3 +97,73 @@ contract HelloWorld {
...
+
+| Option                            | Default  | Description          
+| --------------------------------- | -------  | ---------------------
+| line_length                       | 120      | Maximum line length where formatter will try to wrap the line
+| tab_width                         | 4        | Number of spaces per indentation level
+| bracket_spacing                   | false    | Print spaces between brackets
+| int_types                         | long     | Style of uint/int256 types. Available options: `long`, `short`, `preserve`
+| func_attrs_with_params_multiline  | true     | If function parameters are multiline then always put the function attributes on separate lines
+| quote_style                       | double   | Style of quotation marks. Available options: `double`, `single`, `preserve`
+| number_underscore                 | preserve | Style of underscores in number literals. Available options: `remove`, `thousands`, `preserve`
```
Comment (gakonst) [inline fmt/README.md:113]: I think these might not all exist in the docs? https://book.getfoundry.sh/reference/config?highlight=fmt#formatter

**Review by gakonst — COMMENTED** [review]
good start - do you think the existing docs on the Visitors are good enough? should we have more info on how to use the low-level formatting API?

File: fmt/README.md (hunk 1 above)
Comment (rkrasiuk) [reply fmt/README.md:113]: yup, `int_types`, `number_underscore` & `func_attrs_with_params_multiline` are missing

**Comment by rkrasiuk** [comment]
@gakonst initially, i didn't come up with anything to add without writing a two-page essay with the visitor spec. my thinking is let's get first contributors, consolidate the frequently asked questions and add them to the doc

File: fmt/README.md (hunk 1 above)
Comment (rkrasiuk) [reply fmt/README.md:113]: @gakonst fixed https://github.com/foundry-rs/book/pull/534/files

**Pull request merged**